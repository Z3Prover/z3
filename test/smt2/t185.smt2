
; Copyright (c) 2015 Microsoft Corporation


(set-option :pp.max-depth 100)
(declare-fun p (Int) Bool)
(declare-fun f (Int Real) Bool)

(push)
(assert (not (forall ((x Int) (z Real) (z2 Int)) (=> (and (f x z) (f z2 z)) (exists ((y Int)) (and (not (= y x)) (f y z)))))))
(apply nnf)
(pop)


(push)
(assert (not (forall ((x Int) (z Real) (z2 Int)) (! (=> (and (f x z) (f z2 z)) (exists ((y Int)) (and (not (= y x)) (f y z)))) :skolemid foo))))
(apply nnf)
(pop)
