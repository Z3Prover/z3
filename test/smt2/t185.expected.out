(goals
(goal
  (f x!2 z!1)
  (f z2!0 z!1)
  (forall ((y Int)) (or (= y x!2) (not (f y z!1))))
  :precision precise :depth 1)
)
(goals
(goal
  (f x!foo!5 z!foo!4)
  (f z2!foo!3 z!foo!4)
  (forall ((y Int)) (or (= y x!foo!5) (not (f y z!foo!4))))
  :precision precise :depth 1)
)
